### PR TITLE
RTEMIS-59: Modified the query for school udise code view.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-59

---
### What the PR does
- District admin will only get those UDISE codes for approval that fall under his or her district.

### CODE REVIEW

File: modules/rte_mis_state/rte_mis_state.module [hunk 1]
```diff
@@ -293,3 +296,35 @@ function rte_mis_state_form_user_form_alter(&$form, &$form_state, $form_id) {
...
+      $definition = [
+        'table' => 'taxonomy_term__field_location',
+        'field' => 'entity_id',
+        'left_table' => 'taxonomy_term_field_data',
+        'left_field' => 'tid',
+        'type' => 'INNER',
+      ];
+      $join = Drupal::service('plugin.manager.views.join')->createInstance('standard', $definition);
+      $query->addRelationship('taxonomy_term__field_location', $join, 'taxonomy_term_field_data');
+      $curr_user_location = $curr_user->get('field_location_details')->getValue()[0]['target_id'];
```
Comment (kunal-sin) [inline modules/rte_mis_state/rte_mis_state.module:318]: Use `->getString()` function to get the target id

Comment (kunal-sin) [inline modules/rte_mis_state/rte_mis_state.module:318]: Please add proper comment explaining the reason and code

Comment (RohanSingh-inno) [reply modules/rte_mis_state/rte_mis_state.module]: Ok.

File: modules/rte_mis_state/rte_mis_state.module [hunk 2]
```diff
@@ -293,3 +298,55 @@ function rte_mis_state_form_user_form_alter(&$form, &$form_state, $form_id) {
...
+/**
+ * Implements hook_user_login().
+ */
+function rte_mis_state_user_login(AccountInterface $account) {
+  // Check if the user has the 'district_admin' role.
+  if ($account->hasRole('district_admin')) {
+    // Invalidate the cache of the "school_udise_code" view.
+    Cache::invalidateTags(['config:views.view.school_udise_code']);
+  }
+}
```
Comment (kunal-sin) [inline modules/rte_mis_state/rte_mis_state.module:352]: What will happen when a new UDISE is added for review and the district admin is already logged in?

Comment (RohanSingh-inno) [reply modules/rte_mis_state/rte_mis_state.module:352]: For this scenario, the data gets updated if the district_admin refreshes the page.

**Pull request merged**